coreos-installer

```diff
@@ -613,9 +613,23 @@ log() {
 #########################################################
 write_image_to_disk() {
     log "Writing disk image"
-    
+
+    set -o pipefail
+    # Check for gzip and xz correspondingly
+    if [[ $(xxd -ps -l 2 /mnt/dl/imagefile.gz) = '1f8b' ]]
```

Comment:
`xxd` doesn't exist on every system and doesn't exist in the initramfs.  But this should be possible with tools that do, e.g.:

```
[ "$(dd count=2 bs=1 if=/mnt/dl/imagefile.gz)" = "$(printf '\x1f\x8b')" ]
```

Reply:
Yes I was wondering if `xxd` is widely available, will be updating :+1: